decompiler/pipeline/controlflowanalysis/readability_based_refinement.py

```diff
@@ -219,6 +220,28 @@ def _requirement_without_reinitialization(ast: AbstractSyntaxTree, node: Abstrac
                 return True
 
 
+def removeGuardedDoWhile(ast: AbstractSyntaxTree):
+    """ Removes a if statement which guards a do-while loop when:
+            -> there is nothing in between the if-node and the do-while-node 
+            -> the if-node has only one branch
+            -> the condition of the branch is the same as the condition of the do-while-node
+    """
+    for loop_node in list(ast.get_loop_nodes_post_order()):
+        if not isinstance(loop_node, DoWhileLoopNode) or not isinstance(loop_node.parent.parent, ConditionNode):
+            continue
+
+        condition_node = loop_node.parent.parent
+        if len(condition_node.children) != 1:
+            continue
+
+        if condition_node.true_branch:
```

Comment:
This should be unnecessary. You run the `clean_up` before you start this algorithm. This procedure guarantees you that a Condition Node has always a True-Branch.
--> `clean_up` calls `clean` for each node and the `clean` function for condition-nodes handles this (https://github.com/fkie-cad/dewolf/blob/5fec2125493d026b7d8796bcabc41302f5bf6f3d/decompiler/structures/ast/ast_nodes.py#L553)

Reply:
Ah i see.
Changed the code so:
- condition nodes with 2 branches will still be skipped 
- the condition comparison to the condition node itself (which is the same as the `condition_node.true_branch.branch_condition`